EIPS/eip-5143.md

```diff
@@ -0,0 +1,224 @@
+MUST revert if withdrawing `assets` underlying tokens requires burning more then `maxShares` shares.
+
+Note that some implementations will require pre-requesting to the Vault before a withdrawal may be performed. Those methods should be performed separately.
+
+```yaml
+- name: withdraw
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: assets
+      type: uint256
+    - name: receiver
+      type: address
+    - name: owner
+      type: address
+    - name: maxShares
+      type: uint256
+
+  outputs:
+    - name: shares
+      type: uint256
+```
+
+#### redeem
+
+Overloaded version of ERC-4626's `redeem`.
+
+Burns exactly `shares` from `owner` and sends `assets` of underlying tokens to `receiver`.
+
+MUST emit the `Withdraw` event.
+
+MUST support a redeem flow where the shares are burned from `owner` directly where `owner` is `msg.sender` or `msg.sender` has ERC-20 approval over the shares of `owner`.
+MAY support an additional flow in which the shares are transferred to the Vault contract before the `redeem` execution, and are accounted for during `redeem`.
+
+MUST revert if all of `shares` cannot be redeemed (due to withdrawal limit being reached, slippage, the owner not having enough shares, etc).
+MUST revert if redeeming `shares` shares sends less than `minAssets` underlying tokens to `receiver`.
+
+Note that some implementations will require pre-requesting to the Vault before a withdrawal may be performed. Those methods should be performed separately.
+
+```yaml
+- name: redeem
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: shares
+      type: uint256
+    - name: receiver
+      type: address
+    - name: owner
+      type: address
+    - name: minAssets
+      type: uint256
+
+  outputs:
+    - name: assets
+      type: uint256
+```
+
+## Rationale
+
+This ERC's functions do not replace ERC-4626 equivalent mechanisms. They are additional (overloaded) methods designed to protect EOAs interacting with the vault.
+
+When smart contracts interact with an ERC-4626 vault, they can preview any operation using the dedicated functions before executing the operation. This can be done
+atomically, with no risk of price change. This is not true of EOA, which will preview their operations on a UI, sign a transaction, and have it mined later.
+Between the preview and the transaction being executed, the blockchain state might change, resulting in unexpected outcomes. In particular, frontrunning
+make EOA's interractons with an ERC-4626 vault possibly risky.
+
+Other projects in the DeFi spaces, such as decentralized exchanges, already include similar mechanisms so a user can request its transaction reverts if the
+resulting exchange rate is not considered good enough.
+
+Implementing This ERC on top of an ERC-4626 contract can be done very easily. It just requires calling the corresponding ERC-4626 function and adding a revert
+check on the returned value.
+
+## Alternative approaches
```

Comment:
Not a valid top level section; I'd recommend putting it inside the Rationale section (so `### Alternative Approaches`.)